common/persistence/visibility/visiblity_manager_metrics.go

```diff
@@ -137,7 +141,14 @@ func (m *visibilityManagerMetrics) ListWorkflowExecutions(
 ) (*manager.ListWorkflowExecutionsResponse, error) {
 	handler, startTime := m.tagScope(metrics.VisibilityPersistenceListWorkflowExecutionsScope)
 	response, err := m.delegate.ListWorkflowExecutions(ctx, request)
-	metrics.VisibilityPersistenceLatency.With(handler).Record(time.Since(startTime))
+	elapsed := time.Since(startTime)
+	if elapsed > m.slowQueryThreshold() {
+		m.logger.Info("List query exceeded threshold",
+			tag.NewDurationTag("duration", elapsed),
+			tag.NewStringTag("visibility-query", request.Query),
+		)
```

Comment:
Can we do warn level instead of info?
Also, can we tag namespace?

Reply:
Yes. Let me see how available namespace is.

Reply:
Added both suggestions.